pkg/scheduler/framework/interface.go

```diff
@@ -538,6 +538,9 @@ type Handle interface {
 
 	// TODO: unroll the wrapped interfaces to Handle.
 	PreemptHandle() PreemptHandle
+
+	// ProfileName returns a profile name.
+	ProfileName() string
```

Comment:
Move this to the Profile interface and just call it Name

Reply:
+1.

Reply:
I think if move this to the `Profile` interface,  `framework.Framework` interface cannot be used as `Profile`.

Reply:
Discussed offline: we would need a cast for now.

In a follow up, we should make `runtime.NewFramework` return a pointer to an exported struct, instead of returning a `framework.Framework`. @Huang-Wei do you agree?

Reply:
Casting sounds an smell to me.

> In a follow up, we should make runtime.NewFramework return a pointer to an exported struct, instead of returning a framework.Framework. @Huang-Wei do you agree?

Better not as it would sacrifice the refactoring work we've done so far on interface abstractions.

---

Can we merge Profile and Framework into one interface? Currently we just instantiate `frameworkImpl` to represent Framework and also Profile, and sort of use these 2 interfaces interchangeably:

```go
# Use a FrameworkFactory to build a Profile.
func NewProfile(cfg config.KubeSchedulerProfile, frameworkFact FrameworkFactory, recorderFact RecorderFactory,

# While pass in a KubeSchedulerProfile to build a FrameworkFactory.
type FrameworkFactory func(config.KubeSchedulerProfile, ...frameworkruntime.Option) (framework.Framework, error)

# Compose profiles while using buildFramework
profiles, err := profile.NewMap(c.profiles, c.buildFramework, c.recorderFactory,
```

The semantics would make more sense like this:

```go
func NewProfile(cfg config.KubeSchedulerProfile, profileFact ProfileFactory, recorderFact RecorderFactory,
type ProfileFactory func(config.KubeSchedulerProfile, ...frameworkruntime.Option) (framework.Profile, error)
profiles, err := profile.NewMap(c.profiles, c.buildProfile, c.recorderFactory,
```

Reply:
I would prefer not to rename `framework.Framework` to `framework.Profile`, but yes, we could simply get rid of `profile.Profile`. And `NewProfile` could simply be unexported, right?

Reply:
> And `NewProfile` could simply be unexported, right?

I think so, it's only used in `profile.go#NewMap`.

> I would prefer not to rename `framework.Framework` to `framework.Profile`, but yes, we could simply get rid of `profile.Profile`.

Are you concerned with the refactoring efforts, or(and) the downstream developers who have used `framework.NewFramework()` a lot?

I'm ok with holding it as I don't want to break the users, although "Profile" is a more self-explanatory term IMO. So what's our plan here: get rid of `profile.Profile` and move `ProfileName()` to `framework.Framework`?

Reply:
I'm more concerned with downstream devs.

Yes, that would be the plan.

Reply:
So I will get rid of `profile.Profile` and use `framework.Framwork` instead of `profile.Profile`.